Modules/_ctypes/callproc.c

```diff
@@ -1990,14 +1990,23 @@ create_pointer_type(PyObject *module, PyObject *cls)
 {
     PyObject *result;
     PyTypeObject *typ;
-    PyObject *key;
 
     assert(module);
     ctypes_state *st = get_module_state(module);
+    StgInfo* info = NULL;
+    if (PyStgInfo_FromAny(st, cls, &info) < 0) {
+        return NULL;
+    }
+
+    if (info && info->pointer_type) {
+        return Py_XNewRef(info->pointer_type);
+    }
```

Comment:
```suggestion
    StgInfo *info = NULL;
    if (PyType_Check(cls)) {
        if (PyStgInfo_FromType(st, cls, &info) < 0) {
            return NULL;
        }
        if (info && info->pointer_type) {
            return Py_NewRef(info->pointer_type);
        }
    }
```

* `create_pointer_type()` is supposed to reject ctypes _objects_.  The existing `PyType_Check(cls)` conditional branch could be merged.

    ```
    >>> POINTER(c_int)     # cached
    >>> POINTER(c_int(1))  # returns cache (not allowed)
    ```

* `Py_NewRef()` suffices when `info->pointer_type` is non-NULL.

Reply:
Thanks. Especially for `POINTER(c_int(1))`.